stainless_extraction/src/fns.rs

```diff
@@ -58,3 +58,17 @@ impl<'a> FnItem<'a> {
     self.span.from_expansion() && self.spec_type.is_some()
   }
 }
+
+/// Identifies the specific implementation/instance of a type class that is
+/// needed at a method call site.
+///
+/// 1) If this is a trait method, then the identifier is the id of the trait where
+/// the method is defined. Otherwise, of the impl where it is defined.
+///
+/// 2) The type is the receiver type (aka the type of the first argument at call site).
+/// 3) The other types are optional type parameters of the receiver type.
+pub type TypeClassKey<'l> = (
```

Comment:
Feel free to ignore if this is a convention, or if it makes the pattern match in `extract_method_receiver` to difficult to read (though that might be a sign that this code could be expressed differently?).

Reply:
I think that's a good idea. As said, the handling of multiple type parameters is not yet mature at all. So, in the future we might even merge the two type fields to a single type vector.

Reply:
Or should I do it right away, as I find myself splitting the type vectors I receive from elsewhere?

Reply:
From a type theoretic POV, `(A, B, Vec<B>)` is isomorphic to `(A, NonEmptyVec<B>)`, so if you can guarantee the latter (which I guess you can since you always have a receiver type) then I don't see a problem with merging the two fields.